libraries/libfile/ZipArchive.cpp

```diff
@@ -253,7 +245,12 @@ void ZipArchive::write_entry(const Entry &entry, BinaryWriter &writer, const Vec
 
     writer.put(header);
     writer.put_fixed_len_string(entry.name);
-    writer.put_data(compressed_data.raw_storage(), compressed_data.count());
+
+    uint8_t *data_buffer = nullptr;
+    size_t data_size = 0;
+    compressed.read_all((void **)&data_buffer, &data_size);
+    writer.put_data(data_buffer, data_size);
+    delete[] data_buffer;
```

Comment:
Isn't this a use case for copyto or copyfrom ?

Reply:
Yeah it should be, but **BinaryWriter** is no real writer yet. I'll fix it, just leave it open

Reply:
ok :+1: